HashMapSort/src/main/java/com/AndrieievYurii/ElementsDistribution.java

```diff
@@ -0,0 +1,36 @@
+package com.AndrieievYurii;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class ElementsDistribution {
+
+    public HashMap<Character, Integer> elementDistribution(String input) {
+        HashMap<String,HashMap<Character,Integer>> words = new HashMap<>();
```

Comment:
Never use a particular implimentation in variable Declaration!! Use always ancestor instead. HashMap is implimentation of interface Map. So right Declaration is 
Map<String, Map<Character, Integer>> words = new HashMap<>();
Or you forgot about Polymorphism principle in OOP

Reply:
And name of this variable can be better - this is realization of you cache - so maybe cachedWords?